pkg/manager/member/dm_worker_member_manager.go

```diff
@@ -259,6 +259,19 @@ func (wmm *workerMemberManager) syncDMClusterStatus(dc *v1alpha1.DMCluster, set
 		}
 
 		workerStatus[name] = status
+
+		// offline the workers that already been scaled-in
+		if status.Stage == "offline" {
+			ordinal, err := util.GetOrdinalFromPodName(worker.Name)
+			if err != nil {
+				klog.Errorf("invalid worker name %s, can't offline this worker automatically, err: %s", worker.Name, err)
+			} else if ordinal >= dc.WorkerStsDesiredReplicas() {
+				err := dmClient.DeleteWorker(name)
```

Comment:
This should be done in the `scaleIn` function?

Reply:
dm-worker can't be offlined until the keepalive lease in `dm-master` is outdated. If we do this in `scaleIn` function it will possibly fail.

Reply:
What if we deleteWorker first before deleting the Pod in the scaleIn function?